### ISSUE TITLE
Gogs installation keeps creating table with collation utf8mb4_general_ci even though the database has the collation utf8_general_ci

### ISSUE THREAD

**Issue by tostiheld** [gogits/gogs#5063]
- Gogs version (or commit ref): 0.11.34.1122
- Git version: 2.11.0
- Operating system: Debian Stretch (Linux 4.9.0-6-amd64)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [x] No
  - [ ] Not relevant
- Log gist (usually found in `log/gogs.log`): no log found

## Description

Whenever I run the installation on my setup (MySQL is actually Mariadb by the way), the setup keeps failing because the index for some tables get too large. Which I understand and that's why I set the database collation for gogs to utf8_general_ci. But the installation seems to ignore that, and keeps creating tables in the utf8mb4_general_ci collation anyways.

**Comment by 4nd3r** [comment]
with mysql you should always use utf8mb4 and not utf8, if possible.

see: https://www.google.com/search?q=utf8mb4+vs+utf8

tldr
> "utf8" cannot store all Unicode code points

**Comment by tostiheld** [comment]
To be honest I don't care about storing every possible utf code point, I just wanted a working gogs installation. Switched to gitea anyways.

**Comment by 4nd3r** [comment]
no emojis for you, then :laughing:

**Comment by yogicodes** [comment]
Still no emoji support! on Description, Lables etc 😢
